packages/web3wallet/src/client.ts

```diff
@@ -0,0 +1,183 @@
+import EventEmitter from "events";
+import { CLIENT_CONTEXT } from "./constants";
+import { Engine, PendingRequest } from "./controllers";
+import { IWeb3Wallet, Web3WalletTypes } from "./types";
+
+export class Web3Wallet extends IWeb3Wallet {
+  public name: IWeb3Wallet["name"];
+  public core: IWeb3Wallet["core"];
+  public logger: IWeb3Wallet["logger"];
+  public events: IWeb3Wallet["events"] = new EventEmitter();
+  public engine: IWeb3Wallet["engine"];
+  public pendingRequest: IWeb3Wallet["pendingRequest"];
+  public metadata: IWeb3Wallet["metadata"];
+
+  static async init(opts: Web3WalletTypes.Options) {
+    const client = new Web3Wallet(opts);
+    await client.initialize();
+
+    return client;
+  }
+
+  constructor(opts: Web3WalletTypes.Options) {
+    super(opts);
+    this.metadata = opts.metadata as any; // TODO: automatic fetch of metadata when?
```

Comment:
Why the cast to `any` here?

Can probs remove the todo that is actually a question too 😄

Reply:
making it required as it's required in `Auth`